stunnel/stunnel.ml

```diff
@@ -144,6 +144,15 @@ let config_file verify_cert extended_diagnosis host port legacy =
       bool_of_string (Inventory.lookup ~default:"false" "CC_PREPARATIONS")
     with _ -> false
   in
+  let is_valid_crl_path crl_path =
+    try match Sys.is_directory crl_path with
+      | true ->
+        (match Sys.readdir crl_path with
+         | [| |] -> false
+         | _ -> true)
+      | false -> false
```

Comment:
I think the code could be clearer if it was expressed as
```
if is_dir && not_empty readdir then
  true
else
  false
```

Reply:
I'd do this:

```
match Sys.readdir crl_path with
| [| |] -> ..
| _ ->  ..
| exception _ -> ..
```

Reply:
Just tested it. It works amazingly. 
I had a concern on the order of `_` and `exception`. It proved that `exception` would not be caught by the `_`.

Reply:
BTW, I can't combine the `| [| |]` and `| exception _` in one line as they share same result.
```
Error: Exception patterns must be at the top level of a match case.
```

Reply:
Yes `exception` always has to be separate I think, the default wildcard `_` only catches values, not exceptions.